aten/src/ATen/native/Linear.cpp

```diff
@@ -91,7 +98,35 @@ Tensor linear(const Tensor& input, const Tensor& weight, const std::optional<Ten
     return xnnpack::linear(input, weight, *bias);
   }
 #endif
-  if (input_dim == 2 && bias->defined()) {
+
+  // Weight has been prepacked
+  if (weight.is_mkldnn()) {
```

Comment:
Can we reuse `at::kldnn_linear` or improve it?

Reply:
Then, does it make sense to revise `mkldnn_linear` to use the matmul primitive instead? cc @yanbing-j 
My concern is the complexity added to the addmm, which seems not necessary since working from `linear` sounds good enough to me. If "inner_product" is currently slower, you can perhaps add a dedicated path using matmul primitives for aarch64 in `mkldnn_linear`. Later on, we can further simplify the implementation of `mkldnn_linear` to always use the matmul primitive, suppose it is also faster on x86.

Reply:
Done.